joinservice/internal/watcher/validator.go

```diff
@@ -76,11 +79,30 @@ func (u *Updatable) Update() error {
 	}
 	u.log.Debugf("New expected measurements: %+v", cfg.GetMeasurements())
 
-	validator, err := choose.Validator(cfg, u.log)
+	cfgWithCerts := u.addCachedCerts(cfg)
+
+	validator, err := choose.Validator(cfgWithCerts, u.log)
 	if err != nil {
-		return fmt.Errorf("updating validator: %w", err)
+		return fmt.Errorf("choosing validator: %w", err)
 	}
 	u.Validator = validator
 
 	return nil
 }
+
+// addCachedCerts adds the certificates cached by the validator to the config, if applicable.
+func (u *Updatable) addCachedCerts(cfg config.AttestationCfg) config.AttestationCfg {
+	if u.cachedCerts != nil {
+		// SEV-SNP (ASK and ARK) Certificates
+		ask, _ := u.cachedCerts.SevSnpCerts()
+		if ask != nil {
+			switch c := cfg.(type) {
+			case *config.AzureSEVSNP:
+				c.AMDSigningKey = config.Certificate(*ask)
+				return c
+			}
+			// TODO(derpsteb): Add AWS SEV-SNP
+		}
+	}
+	return cfg
+}
```

Comment:
```suggestion
// addCachedCerts adds the certificates cached by the validator to the config, if applicable.
func (u *Updatable) addCachedCerts(cfg config.AttestationCfg) (config.AttestationCfg, error) {
	switch c := cfg.(type) {
	case *config.AzureSEVSNP:
		ask, err := u.getCachedCerts()
		if err != nil {
			return nil, err
		}
		c.AMDSigningKey = config.Certificate(ask)
		return c, nil
	}
	// TODO(derpsteb): Add AWS SEV-SNP

	return cfg, nil
}

func (u *Updatable) getCachedCerts() (x509.Certificate, error) {
	if u.cachedCerts == nil {
		return x509.Certificate{}, fmt.Errorf("no cached certs available")
	}
	ask, ark := u.cachedCerts.SevSnpCerts()
	if ask == nil {
		return x509.Certificate{}, fmt.Errorf("no ASK available")
	}
	if ark == nil {
		return x509.Certificate{}, fmt.Errorf("no ARK available")
	}
	return *ask, nil
}
```
I think frontloading the attestation variant check makes this function a bit easier to read and maintain. It also handles errors better (cached certs not set, or certificates missing)